internal/uvm/vsmb.go

```diff
@@ -107,6 +108,22 @@ func (uvm *UtilityVM) AddVSMB(ctx context.Context, hostPath string, guestRequest
 			},
 			ResourcePath: vSmbShareResourcePath,
 		}
+
+		if guestRequest != "" {
```

Comment:
I don't really like making this function take a guest request argument. Could we issue the guest request separately instead?

Reply:
I found it odd as well, especially since we just end up passing "" for the guestrequest everywhere this is currently called before this change.

Reply:
@kevpar Was thinking about this, if the only guest request we're going to do is a mapped directory one for VSMB, should we construct it inside and then just have a bool argument called mapToPath or similar instead of passing in a guest request?

Reply:
I think we should just make it so `VSMBShare` exposes the UVM share path, and then do the guest request in the caller instead of in here. I don't see any reason to tie the guest request to the act of doing the VSMB mapping.

Reply:
We can do 
```go
func (vsmb *VSMBShare) GuestPath() string {
    return vsmb.guestPath
}
```
and build the guest request with this but uvm.modify isn't exported. How about a method on VSMBShare called MapToPath or similar?

Reply:
Ideally I don't think the VSMB code should have anything to do with sending a MappedDirectory request inside of the guest. Those should be separate operations. For instance once we move to a new interface for shim-virtstack interactions, we will need these to be decoupled. Can we add a new exported method on `UtilityVM` that allows sending a guest request, and use that?

Reply:
Yep, I didn't know if we wanted to expose something like that as we have modify unexported to begin with. I'll do that then